Adding support for translating currency symbols

douglasmiller opened the pull request:
This will allow currency symbols to be configured on a per-locale basis.

ggilder commented on lib/money/money/formatting.rb:
```diff
@@ -343,6 +344,15 @@ def regexp_format(formatted, rules, decimal_mark, symbol_value)
 
+  def translate_formatting_rules(rules)
+    begin
+      rules[:symbol] = I18n.t currency.iso_code, :scope => "number.currency.symbol", :raise => true
```
This pattern seems pretty expensive in a situation where no translations are used. Can this entire behavior be made optional, as it seems like it's probably not the majority use case?

douglasmiller replied:
Ah good point. I'll add a `:translate` options to the rules hash

semmons99 commented:
seems like a good candidate for a toggle if you're not using the feature.

douglasmiller commented:
I've added a `:translate` option to the rules hash and documented the feature.

I also updated other documentation that was erroneous:
```
Money.new(100, "USD").format -> $100.00
```

bradfeehan commented:
I have a use for this, it'd be great to see it merged. Otherwise, is there a way to show "US$" for USD only when the locale isn't `en` or `en_US`? It doesn't have a disambiguation column.

semmons99 commented:
Sorry, I thought I had.
